src/controller/src/recording_controller.cpp

```diff
@@ -0,0 +1,17 @@
+#include "controller/recording_controller.h"
+
+using namespace controller;
+
+RecordingController::RecordingController(std::string json_file_name) {
+	Controller();
```

Comment:
Will this not be called implicitly? Is this call needed?

Reply:
Yup, it actually will be called by the compiler implicitly while creating that `base class` -`sub class` relation. It's just an old habit I guess, which I picked up from my JAVA class the previous semester. :P